sample_annotator/text_mining/TextMining.py

```diff
@@ -4,10 +4,85 @@
 import logging
 from nmdc_schema.nmdc import QuantityValue
 import re
+import os
+import configparser
+
+SETTINGS_FILENAME = 'settings.ini'
+PATH = '.'
 
 @dataclass
 class TextMining():
-    ...
+    """
+    Text mining Class
+    """
+    
+
+    def create_settings_file(self, path: str = PATH, ontList: List = ['ENVO']) -> None: 
+        """
+        Dynamically creates the settings.ini file for OGER to get parameters.
+
+        :param path: Path of the 'nlp' folder
+        :param ontList: The ontology to be used as dictionary e.g. ['ENVO', 'CHEBI']
+        :return: None.
+
+        -   Include header: Output file will have column names
+
+        -   The 'Shared' section declares global variables that can be used in other sections
+            e.g. Data root.
+            root = location of the working directory
+            accessed in other sections using => ${Shared:root}/
+
+        -   Input formats accepted:
+            txt, txt_json, bioc_xml, bioc_json, conll, pubmed,
+            pxml, pxml.gz, pmc, nxml, pubtator, pubtator_fbk,
+            becalmabstracts, becalmpatents
+
+        -   Two iter-modes available: [collection or document]
+            document:- 'n' input files = 'n' output files
+            (provided every file has ontology terms)
+            collection:- n input files = 1 output file
+
+        -   Export formats possible:
+            tsv, txt, text_tsv, xml, text_xml, bioc_xml,
+            bioc_json, bionlp, bionlp.ann, brat, brat.ann,
+            conll, pubtator, pubanno_json, pubtator, pubtator_fbk,
+            europepmc, europepmc.zip, odin, becalm_tsv, becalm_json
+            These can be passed as a list for multiple outputs too.
+
+        -   Multiple Termlists can be declared in separate sections
+            e.g. [Termlist1], [Termlist2] ...[Termlistn] with each having
+            their own paths
+
+        -   Normalization used: lowercase and stem-Porter
+        """
+
+        config = configparser.ConfigParser()
+        config['Section'] = {}
+        config['Shared'] = {}
+        
+        # Settings required by OGER
+        config['Main'] = {
+            'include_header' : True,
+            'input-directory' : os.path.join(path,'input'),
+            'output-directory' : os.path.join(path,'output'),
+            'pointer-type' : 'glob',
+            'pointers' : '*.tsv',
+            'iter-mode' : 'collection',
+            'article-format' : 'txt_tsv',
+            'export_format': 'tsv',
+            'termlist_stopwords': os.path.join(path,'stopwords','stopWords.txt'),
```

Comment:
see how the mixs json is loaded

The general pattern is

 - allow code user to specify their own paths
 - get the right defaults with zero configuration

Reply:
> see how the mixs json is loaded

Could you please guide me to a sample mixs.json file?

Reply:
https://github.com/microbiomedata/sample-annotator/blob/bc17c1267b1331e92c33685a9e6cdda47e7ac8e0/sample_annotator/__init__.py#L5